### DISCUSSION TITLE
Notification model raising ActiveJob::SerializationError exception when querying with an active record instance inside params on rails > 7.0.4.3

### DISCUSSION THREAD

**Discussion by Roupiye** [excid3/noticed#323] [General]
RAILS 7.0.5 reproduction:
- clone https://github.com/TopRoupi/noticed_bug_test
- change the database.yml first for the app to work
- run `rails db:create db:migrate` 
- run `rails c`
- then execute the following lines:

```
Comment.new(body: "test").save
Notification.where(params: {comment: Comment.last}).last
```

expected behavior:
`Notification.where(params: {comment: Comment.last}).last` should not raise and return nil
current behavior:
it raises SerializationError

RAILS 7.0.4.3 reproduction:
- clone https://github.com/TopRoupi/noticed_bug_test/tree/1b8e556420a9464991359af13e0fa0f882b023be
- edit the database.yml and migrate the database
- then when running `rails c` and `Notification.where(params: {comment: Comment.last}).last` it wont raise like when using rails 7.0.5

**Comment by drewblas** [comment]
We're having the same issue as well.  Not really seeing a clear way to fix it unfortunately.

**Comment by Roupiye** [comment]
this is my dirty temporary fix:
```ruby
class ApplicationRecord < ActiveRecord::Base
  #TODO: remove this after https://github.com/rails/rails/issues/48652 is fixed
  def deep_dup
    self
  end
end
```
i think an actual fix would be to make .deepdup make an actual deep copy(that will also copy the object references see https://apidock.com/ruby/Object/dup)

https://github.com/rails/rails/blob/a5fc471b3f4bbd02e6be38dae023526a49e7d049/activesupport/lib/active_support/core_ext/object/deep_dup.rb#L15
deepdup is currently just calling dup
